Tracking/components/TracksFromGenParticles.cpp

```diff
@@ -101,35 +101,62 @@ struct TracksFromGenParticles final
     // - barrel: inner R, zmax
     // - endcap: inner R, zmin, zmax
     if (m_extrapolateToECal) {
+      bool retrieved = false;
+
       try {
-        const dd4hep::rec::LayeredCalorimeterData * eCalBarrelExtension = getExtension( ( dd4hep::DetType::CALORIMETER | dd4hep::DetType::ELECTROMAGNETIC | dd4hep::DetType::BARREL),
-                                                                                        ( dd4hep::DetType::AUXILIARY  |  dd4hep::DetType::FORWARD ) );
-        m_eCalBarrelInnerR = eCalBarrelExtension->extent[0] / dd4hep::mm;
-        m_eCalBarrelMaxZ = eCalBarrelExtension->extent[3] / dd4hep::mm;
-        debug() << "ECAL barrel extent: Rmin [mm] = " << m_eCalBarrelInnerR << endmsg;
-        debug() << "ECAL barrel extent: Zmax [mm] = " << m_eCalBarrelMaxZ << endmsg;
+        const dd4hep::rec::LayeredCalorimeterData * caloExtension = getExtension( ( dd4hep::DetType::CALORIMETER | dd4hep::DetType::BARREL | dd4hep::DetType::ENDCAP),
```

Comment:
Does this always throw when it's not found? If not we could check the returned pointer which should be `nullptr`? The several try - except look a bit ugly